app/src/main/java/com/example/playlistmakervmp/SearchActivity.kt

```diff
@@ -177,6 +228,66 @@ class SearchActivity : AppCompatActivity() {
             placeholderNothingFoundText.visibility = View.GONE
             searchTracks()
         }
+
+        trackAdapter.setOnTrackClickListener(object : OnTrackClickListener {
+            override fun onTrackClick(position: Int) {
+                val editor = sharedPrefs.edit()
+                if ((history.size <= 9) and !isTrackInHistory(tracks[position])) {
+                    history.add(0, tracks[position])
+                    json = Gson().toJsonTree(history).asJsonArray.toString()
+                    editor.putString("TRACKS", json).apply()
+                }
+                else if ((history.size == 10) and !isTrackInHistory(tracks[position])) {
```

Comment:
Число 10 лучше вынести в константу с говорящим названием, чтобы людям, читающим твой код, было проще понимать его.
В твоём случае:
`
const val MAX_HISTORY_SIZE = 10
`
в 240 строке:
```suggestion
                else if ((history.size == MAX_HISTORY_SIZE) and !isTrackInHistory(tracks[position])) {
```
В 235 строке:
`
if ((history.size < MAX_HISTORY_SIZE) and !isTrackInHistory(tracks[position])) {
`